.github/workflows/main.yml

```diff
@@ -51,7 +51,7 @@ jobs:
         run: |
           python setup.py bdist_wheel --skip-build
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         if: ${{ always() }}
         with:
           name: artifacts
```

Comment:
```suggestion
          name: artifacts-py${{ matrix.python-version }}-arch_${{ matrix.architecture }}
```

Reply:
It might be preferable to have all the artefacts under a singular archive to make downloading for deployments easier (until if/when the workflow is updated for automated publishing).

I know v4 no longer merges artefacts by default. I'll have to update these steps.